File: UnityFunctional/UnityFunctional/Resources.fs
namespace UnityFunctional

open UnityEngine
open System

type Resources() =
    inherit MonoBehaviour()

    [<SerializeField>]
    let mutable DildSild:int = 32
    [<SerializeField>]
    let mutable MildAbild:int = 45
    [<SerializeField>]
    let mutable Vildild:int = 5
    [<DefaultValue>]
    val mutable ResourceController:ResourceController

    member this.Start() =
        this.ResourceController.UpdateResources DildSild MildAbild Vildild
        this.ResourceController.UpdateResults 0 0 0

File: Assets/Resources/Resources.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

public class Resources : MonoBehaviour
{
    public int DildSild;
    public int MildAbild;
    public int Vildild;
    public ResourceController ResourceController;

    // Start is called before the first frame update
    void Start()
    {
        ResourceController.UpdateResources(DildSild, MildAbild, Vildild);
        ResourceController.UpdateResults(0, 0, 0);
    }
}
